### ISSUE TITLE
Theming, part I

### ISSUE THREAD

**Issue by ludwiczakpawel** [forem/forem#14835]
1. Remove following themes: "pink", "minimal", "hacker" and leave only "default" and "night".
2. Rename themes: "default" --> "light", "night" --> "dark".
3. Add option (literally a checkbox) to follow user system's `prefers-color-scheme` preference ("_Sync with OS settings_") https://github.com/forem/forem/issues/17702
    - make it "checked" by default 
    - it should work for logged out users too
4. Migrate users – if they currently have:
    - “night” or “hacker” - we should default them to “dark” (“Sync with OS settings” **unchecked**)
    - “minimal” - we should default them to “light” (“Sync with OS settings” **unchecked**)
    - “pink” or “default” - we should default them to “Sync with OS settings”.
5. Let admins choose accent color for the dark theme too (they can currently choose only one, for default theme).
    - If they don't specify it, we can calculate it based on the default one and applying certain filters to make sure it has proper contrast ratio etc.

---

In terms of UX we should try to mimic Slack's app approach:

![image](https://user-images.githubusercontent.com/108287/135066433-5248276d-7a55-485f-8321-56b02f65d39d.png)

---

Corresponding doc: https://docs.google.com/document/d/13frDxjLeGUHKOP8gmXjbFl3mCy7HPOGS-VXCOERiNdc/edit#heading=h.8qmt33fe134z Solutions 1 & 2.

**Comment by citizen428** [comment]
> Remove following themes: "pink", "minimal", "hacker" and leave only "default" and "night".

Just in case anyone else wonders about the user numbers on DEV:

Default: 616,401
Night: 69,651
Pink: 1,971
Minimal: 7,692
Hacker: 6,202

We probably still should give the ~15k users who use soon-to-be deprecated themes a bit of a headsup.

**Comment by aitchiss** [comment]
> (tbh, I'd rather just default them all to light and send them a notification to make their own choices)

Or I guess if we default them all to "follows your OS setting" then people who are really into dark mode are less likely to need to alter their settings

**Comment by ludwiczakpawel** [comment]
> We probably still should give the ~15k users who use soon-to-be deprecated themes a bit of a heads-up

+1

> we default them all to "follows your OS setting" then people who are really into dark mode are less likely to need to alter their settings

+1 too!

**Comment by citizen428** [comment]
Saw this just now, seems to fit the general discussion and our mission of inclusivity and accessibility:

![Github colorblind schemes announcement](https://user-images.githubusercontent.com/47985/135376583-b097599b-6088-4425-9bfb-dc3ebf6af088.png)

**Comment by citizen428** [comment]
>Or I guess if we default them all to "follows your OS setting" then people who are really into dark mode are less likely to need to alter their settings

Yes, that's a much better idea.

**Comment by ludwiczakpawel** [comment]
> Not saying we need to offer all of the above just for reference/future inspiration

I **personally** think GitHub offers too much when it comes to theming. Both, day and night themes, have bunch of color variants which I don't quite _understand_. But I do admire they fact they're adding color blindness themes and we should do it too eventually. Lets just clean up everything else first :).

**Comment by aitchiss** [comment]
For what it's worth, I think good accessibility is avoiding using colours like red/green to convey information in the first place 😄  Like, it's more inclusive if the main dark/light themes work well for everyone.

It would be nice to consider a "high contrast" theme in the future though, and I do like the idea of creating new themes with purpose, rather than just to "look different" 👍

**Comment by ellativity** [comment]
I'm so sad about this. Using different themes is how I know which Forem I'm on when I'm admin-ing DEV (dark), CodeNewbie (minimal), Forem.dev (hacker), and browsing Forem.team (pink) and all other Forems (default). 

Will users be able to set custom themes in the near future, or are those of us who are polyforem gonna have to use more mental capacity to not get confused where we are?

**Comment by aitchiss** [comment]
I do the same @ellativity! The aim of these changes though is so that we can safely help Forem creators have a bit more control of some colours and theming/branding (e.g. changing header colours), which in the end should (hopefully, hypothetically at this point) do more to help Forems feel individual and easily distinguishable while consistent ✨

**Comment by ludwiczakpawel** [comment]
> I'm so sad about this. Using different themes is how I know which Forem I'm on when I'm admin-ing DEV (dark), CodeNewbie (minimal), Forem.dev (hacker), and browsing Forem.team (pink) and all other Forems (default).
> 
> Will users be able to set custom themes in the near future, or are those of us who are polyforem gonna have to use more mental capacity to not get confused where we are?

Don't worry, ultimate goal is for community admins to set up their own color palette, so they should be able to customize not only logo (current state) but also other colors:

- header background color (that should be the most distinctive difference)
- main CTA (buttons) color
- links colors
- maybe body background color

^ so as a user you won't have that many options to choose from. BUT as admin you should be able to customize your Forem muuuuch more. That should give hopefully the appropriate level of distinction between Forems. It will take some time of course, but this is the first step.

Does that makes you a bit less sad?

**Comment by ellativity** [comment]
Yeah I think this will make Forem Creators really happy. We have had a fair bit of feedback from them about not being able to set their own brand colours as the Forem colours, so this is a good move. 

I imagine that alone will resolve the issue of all Forems looking the same. Thanks so much both for clearing this up! I'm grateful for you taking the time to share this info 🙏

**Comment by ludwiczakpawel** [comment]
@amywtlin i think so. but we may want to prep a more organized and detailed doc/spec for engineers to set more clear expectations and to have better place for discussion? i can write one if you think it's needed.

**Comment by citizen428** [comment]
@ludwiczakpawel 

> it should work for logged out users too

To clarify: for logged-out users we always follow their OS setting but logged-in users get a chance to disable this?

> “pink” or “default” - we should default them to “Sync with OS settings”.

I guess we should still also move "pink" to "light", correct? “Sync with OS settings” is an optional checkbox that overrides the other thing but by itself is not a theme so we also need to make sure we have no "pink" remaining in our DB.

**Comment by ludwiczakpawel** [comment]
> To clarify: for logged-out users we always follow their OS setting but logged-in users get a chance to disable this?

Correct.

> I guess we should still also move "pink" to "light", correct? “Sync with OS settings” is an optional checkbox that overrides the other thing but by itself is not a theme so we also need to make sure we have no "pink" remaining in our DB.

There are two separate settings:

1. "Sync with OS settings" - a checkbox. If it's checked, the latter doesn't matter (will be disabled). But if it's unchecked...
2. You can choose "Light" or "Dark".

So we want to move "pink" to "sync with os settings" so neither "light" or "dark" will be selected - if that makes sense (not sure if I described it understandably).

This logic is literally what Slack does with their desktop app so I highly recommend checking it out - it should clarify the behavior.

**Comment by ludwiczakpawel** [comment]
@ellativity just a heads up: @citizen428 is actively working on implementation. It will still take some time to ship it but it's closer than ever so thought you (and your team) may want to know ahead and maybe prep strategy for announcing this change?

**Comment by citizen428** [comment]
Current status:

**Done**

* [Rename default and night theme](https://github.com/forem/forem/pull/15176)

**In progress**

* [Theme data update script, remove theme choices from UI](https://github.com/forem/forem/pull/15176): as per internal discussion we probably leave the changelog posts on forem.dev and dev.to up for about a week, so this will get merged at a later date. The PR is set to draft to avoid accidental merges, is labeled as "merge by any core" and assigned to Jeremy who will be in charge of getting it merged.
* [Theming cleanup](https://github.com/forem/forem/pull/15237) I already had this code left over from my spike, so I decided to open one more PR to clean up after the previous one. Again this is set to draft to avoid accidental merges, is labeled as "merge by any core" and assigned to Jeremy.

**Todo**

* Anything related to making the new `prefer_os_color_scheme`  setting work. For now, we only store this value which was necessary for the data update script but we don't expose a UI for changing it and are missing the necessary CSS for `prefers-color-scheme`.

**Comment by izakfilmalter** [comment]
Would it be possible for Admins of communities to change the default theme to Night?

**Comment by citizen428** [comment]
@izakfilmalter Would you mind [starting a disscuion](https://github.com/forem/forem/discussions/new) for this in the "Feature Request" category so that our product team can stay on top of this request?

**Comment by amywtlin** [comment]
@citizen428 Did we end up finishing all the tasks for this issue, especially those ones which were in progress or to do in your last comment?

**Comment by citizen428** [comment]
@amywtlin The ones that were marked `In Progress` should be done but as far as I can tell we never did the one in `Todo`.

**Comment by amywtlin** [comment]
@ludwiczakpawel I am reevaluating this one and I am looking to turn the last piece of work into a separate issue. May I understand correctly, from Michael's last update and your doc, that the remaining work is to add a setting for users to decide if they want the color mode to be detected and applied automatically based on their OS settings?

**Comment by ludwiczakpawel** [comment]
@amywtlin I _think_ this is correct (sorry it's been a while, I kinda got out of loop with this one)

**Comment by amywtlin** [comment]
@ludwiczakpawel No worries. I think that's good enough to start with 👍  I'll turn that one into an issue and close this one.

**Comment by ludwiczakpawel** [comment]
FWIW I've just reminded myself we need to fix Onboarding theming before we ship it... Onboarding was never optimized for dark theme (to repro, log in to dev, switch to dark theme and then go to dev.to/onboarding) - i estimate fixing it would be 0.5-1 day of work.

**Comment by mikeydorje** [comment]
Most Part I theming work landed via linked PRs; remaining “sync with OS” scope was split into #17702. Closing this issue in favor of the dedicated follow-up.
